app/models/data_file.rb

```diff
@@ -122,7 +122,8 @@ def possible_sample_types(user = User.current_user)
   end
 
   def related_samples
-    extracted_samples
+    sample_ids = SampleResourceLink.where('resource_type': 'DataFile', 'resource_id': id).pluck('sample_id')
```

Comment:
you should be able to do this through a has_many association

Reply:
Resolved in [9bdc250](https://github.com/seek4science/seek/commit/9bdc25028fee8643082a76d41053465f27661353).